subsys/bluetooth/mesh/Kconfig

```diff
@@ -1584,6 +1584,34 @@ config BT_MESH_SEQ_STORE_RATE
 	  off with a value that's guaranteed to be larger than the last
 	  one used before power off.
 
+config BT_MESH_SETTINGS_WORKQ
+	bool "Store the Bluetooth mesh settings in a separate work queue"
+	default y
+	help
+	  This option enables a separate cooperative thread which is used to
```

Comment:
As far as I know, once the cooperative thread is running, other threads cannot preempt it, so sysworkq and BT RX can only wait ?

Reply:
https://docs.zephyrproject.org/3.1.0/kernel/services/scheduling/index.html#cooperative-time-slicing

Reply:
They can, unless they enter a rescheduling point: https://docs.zephyrproject.org/latest/kernel/services/scheduling/index.html#concepts. You can observe the following stack call when storing settings:
`settings_save_one` -> `nvs_write`/`nvs_delete` -> `flash_write` -> `flash_nrf_write` (`soc_flash_nrf.c`) -> `write_synchronously` -> `nrf_flash_sync_exe` (`soc_flash_nrf_ticker.c`) -> `k_sem_take` (https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/controller/flash/soc_flash_nrf_ticker.c#L225). At this point the thread is put on sleep until write or erase is finished. Other cooperative threads can be scheduled here.